### DISCUSSION TITLE
Merge signalbased *.mf4 with framebased extracted *.mf4

### DISCUSSION THREAD

**Discussion by myrthel** [danielhrisca/asammdf#885] [Q&A]
Hi all,

I have a *.mf4 file, which contains some signalbased logs and some framebased, which i convert to signalbased.
Resulting in two mdf-objects.
I would then like to combine these two into one mdf-object.
I tried it like this:
```python
from asammdf import MDF, Signal
mdf = MDF(r'C:/example.mf4')
databases = {
    "CAN": [\
    ("C:/example.dbc", 1),\
    ],
}
result = mdf.extract_bus_logging(
    database_files=databases,
    version=None,
    prefix="",
    progress=None,
)
filename = "C:/test.mf4"
# first get the signallist from all signals contained in the raw mf4-file
sig_list = mdf.search(pattern='.*', mode='regex', case_insensitive=True)
# get back the signal object
mdf_sb = mdf.filter(sig_list,None,None)
result.append(mdf_sb)
result = result.save(
    dst=filename,
    overwrite=True,
)
```
problem is, that the signals, I get from mdf.search() is a list of signalnames which are not unique
so mdf.filter() fails
is there maybe another way to kind of merge mdf objects?
thanks in advance

**Comment by danielhrisca** [comment]
https://asammdf.readthedocs.io/en/latest/api.html#asammdf.mdf.MDF.stack

**Reply by myrthel** [reply]
thanks for that super fast answer! :pray:
